tag_release.py

```diff
@@ -0,0 +1,554 @@
+    ):
+    if parent_repo_name == "edx/edx-platform":
+        requirements_file = "requirements/edx/github.txt"
+    else:
+        requirements_file = "requirements.txt"
+
+    req_file_url = "https://raw.githubusercontent.com/{parent_repo}/{branch}/{req_file}".format(
+        parent_repo=parent_repo_name,
+        branch=parent_branch_name,
+        req_file=requirements_file,
+    )
+    req_file_resp = session.get(req_file_url)
+    req_file_resp.raise_for_status()
+    req_file_lines = req_file_resp.text.splitlines()
+    # strip lines that are empty, or start with comments
+    relevant_lines = [line for line in req_file_lines
+                      if line and not line.isspace() and not line.startswith("#")]
+
+    # find the line that corresponds to this repo
+    repo_lines = [line for line in relevant_lines if repo_name in line]
+    if not repo_lines:
+        msg = "{repo_name} dependency not found in {parent} repo".format(
+            repo_name=repo_name, parent=parent_repo_name,
+        )
+        raise ValueError(msg)
+    if len(repo_lines) > 1:
+        msg = "multiple {repo_name} dependencies found in {parent} repo".format(
+            repo_name=repo_name, parent=parent_repo_name,
+        )
+        raise ValueError(msg)
+
+    dependency_line = repo_lines[0]
+    # parse out the branch/tag
+    match = REQUIREMENT_RE.search(dependency_line)
+    if match:
+        ref = match.group('ref')
+    else:
+        ref = None
+    if not ref:
+        msg = "no reference found for {repo_name} dependency in {parent} repo".format(
+            repo_name=repo_name, parent=parent_repo_name,
+        )
+        raise ValueError(msg)
+
+    return get_latest_commit_for_ref(repo_name, ref, session=session)
+
+
+def tag_exists_in_repo(tag, repo, session):
+    """
+    Returns a boolean indicating whether the given tag already exists in the
+    given repo.
+    """
+    tag_url = "https://api.github.com/repos/{repo}/git/refs/tags/{tag}".format(
+        repo=repo,
+        tag=tag,
+    )
+    tag_resp = session.get(tag_url)
+    return tag_resp.ok
+
+
+def repos_where_tag_exists(tag, repos, session):
+    return [repo for repo in repos if tag_exists_in_repo(tag, repo, session)]
+
+def confirm_repo_commits(to_tag):
+    """
+    Confirm with the user that these are the correct commits to tag.
+    Returns True if the user has confirmed, False if the user wants to cancel.
+    """
+    if not to_tag:
+        print("Nothing to do.")
+        return False
+
+    for repo_name, commit_info in to_tag.items():
+        print("{repo}: {ref} ({type}) {sha}".format(
+            repo=repo_name,
+            ref=commit_info['ref'],
+            type=commit_info['ref_type'],
+            sha=commit_info['sha'][0:7],
+        ))
+        print("  " + commit_info["message"].splitlines()[0])
+    response = raw_input("Is this correct? [y/N] ")
+    return response.lower() in ("y", "yes", "1")
```

Comment:
I think for the -y option, I'd like to still have the output of what was going to be tagged. So only conditionalize the input part of this function?